### PR TITLE
Small modification so SmartDashboard can monitor network connection status

### PR DESCRIPTION
**Pull request by sipsorcery**
At the moment the SmartDashboard does a one off check of the NetworkTable connection status. As such the icon on the SmartDashboard cannot be relied upon should the status change. This pull request is to add an event to the NetworkTable class so the SmartDashboard can receive notifications when the connection status changes.

### CODE REVIEW

File: src/FRC.NetworkTables/TcpSockets/TcpConnector.cs [hunk 1]
```diff
@@ -26,7 +26,10 @@ private static int ResolveHostName(string hostName, out IPAddress[] addr)
                     // Sockets don't all support IPV6
                     if (ipAddress.AddressFamily == AddressFamily.InterNetwork)
                     {
-                        addresses.Add(ipAddress);
+                        if (!addresses.Contains(ipAddress))
+                        {
+                            addresses.Add(ipAddress);
+                        }
```
Comment (ThadHouse) [inline src/FRC.NetworkTables/TcpSockets/TcpConnector.cs]: This change should not be necessary. GetHostAddresses should not be returning the same IP twice. Were you running into issues?

**Comment by ThadHouse** [comment]
I like the idea of this change, however in order to accept this a complete implementation would be necessary, including a way to remove any listener added, and remove it by the delegate type instead of only by the callback id. I'd also like to see methods for IRemoteConnectionListener added so I can merge the code upstream to WPILib as well.

File: src/FRC.NetworkTables/TcpSockets/TcpConnector.cs (hunk 1 above)
Comment (sipsorcery) [inline src/FRC.NetworkTables/TcpSockets/TcpConnector.cs]: Yes I did run into an issue where the same IP address got added twice and a bunch of SocketException's started showing up. It would seem sensible to make this change since it has zero side effects and can potentially prevent a problem...

Comment (ThadHouse) [inline src/FRC.NetworkTables/TcpSockets/TcpConnector.cs]: Interesting. Can you split this change into a separate PR then? I'd like to merge this separately from the rest of the changes.

**Comment by sipsorcery** [comment]
It would be trivial to add the wrapper method for NtCore.RemoveConnectionListener that would satisfy the listener removal request.

I'm not really sure what you mean by "complete implementation"? My own little was just to get the SmartDashboard connection icon working so I didn't have to puzzle over why it was red when the connection was fine.

**Comment by ThadHouse** [comment]
The completeimplementation would mean an identical API to the instance version, using the same callbacks as well. We do not want the raw callback ID passed outside of the library, unless the user is directly using the NtCore static class. For the static version of the connection listeners, IRemove can be null, but the rest of the information needs to be correct.

**Comment by ThadHouse** [comment]
I added these fixes separately.  Closed by #41
